client/reader/stream/post-lifecycle.jsx

```diff
@@ -123,7 +123,7 @@ export default class PostLifecycle extends React.Component {
 			return <PostPlaceholder />;
 		} else if ( post._state === 'error' ) {
 			return <PostUnavailable post={ post } />;
-		} else if ( includes( this.props.blockedSites, +postKey.blogId ) ) {
+		} else if ( ! post.is_external && includes( this.props.blockedSites, +post.site_ID ) ) {
```

Comment:
This was failing because, on a feed stream, the postKey contains a `feedId` and a `postId` (but not a `blogId`).

I've included the `is_external` check because on some external posts, the `post.site_ID` can contain the feed ID rather than the site ID/blog ID, which is what we're after.

Reply:
👍

Reply:
an aside, but we might want to look at switching to a `Set` for `blockedSites`. Linear lookup could be a problem for folks with lots of blocks. Not convinced those folks exist yet though.

Reply:
You're right - not enough to be a problem, I don't think. Site blocks aren't persisted, and we don't fetch a list of existing blocks at any point, so all we have in state is blocked and unblocked sites from the current session.